core/src/cli_commands/update_pool.rs

```diff
@@ -0,0 +1,629 @@
+                && ballot_box
+                    .additional_registers
+                    .get(NonMandatoryRegisterId::R6)
+                    == Some(&pool_box_hash)
+                && ballot_box
+                    .additional_registers
+                    .get(NonMandatoryRegisterId::R7)
+                    == Some(&reward_tokens.token_id.clone().into())
+                && ballot_box
+                    .additional_registers
+                    .get(NonMandatoryRegisterId::R8)
+                    == Some(&(*reward_tokens.amount.as_u64() as i64).into())
+        })
+        .scan(&mut votes_cast, |votes_cast, ballot_box| {
+            if **votes_cast >= min_votes as u64 {
+                return None;
+            }
+            **votes_cast += *ballot_box.ballot_token().amount.as_u64();
+            Some(ballot_box)
+        })
+        .collect();
+    if votes_cast < min_votes as u64 {
+        return Err(UpdatePoolError::NotEnoughVotes(
+            min_votes as usize,
+            vote_ballot_boxes.len(),
+        ));
+    }
+
+    let pool_box_candidate = make_pool_box_candidate(
+        &new_pool_contract,
+        old_pool_box.rate() as i64,
+        old_pool_box.epoch_counter() as i32,
+        old_pool_box.pool_nft_token(),
+        reward_tokens.clone(),
+        old_pool_box.get_box().value,
+        old_pool_box.get_box().creation_height, // creation info must be preserved
+    )?;
+    let mut update_box_candidate =
+        ErgoBoxCandidateBuilder::new(update_box.get_box().value, update_box.ergo_tree(), height);
+    update_box_candidate.add_token(update_box.update_nft());
+    let update_box_candidate = update_box_candidate.build()?;
+
+    // Find unspent boxes without ballot token, see: https://github.com/ergoplatform/oracle-core/pull/80#issuecomment-1200258458
+    let unspent_boxes: Vec<ErgoBox> = wallet
+        .get_unspent_wallet_boxes()?
+        .into_iter()
+        .filter(|wallet_box| {
+            wallet_box
+                .tokens
+                .as_ref()
+                .and_then(|tokens| {
+                    tokens
+                        .iter()
+                        .find(|token| token.token_id == update_box.ballot_token_id())
+                })
+                .is_none()
+        })
+        .collect();
+    if unspent_boxes.is_empty() {
+        error!("Could not find unspent wallet boxes that do not contain ballot token. Please move ballot tokens to another address");
+        return Err(UpdatePoolError::NoUsableWalletBoxes);
+    }
+
+    let target_balance = BoxValue::SAFE_USER_MIN;
+    let tokens_needed; // Amount of reward tokens we need from wallet box for new pool box
+    let target_tokens: &[Token] = if reward_tokens.token_id != old_pool_box.reward_token().token_id
+    {
+        tokens_needed = [reward_tokens.clone()];
+        &tokens_needed
+    } else if reward_tokens.amount > old_pool_box.reward_token().amount {
```

Comment:
I don't think this case is real. Since all reward tokens are locked in the pool box upon bootstrap, the only way one can increase the reward token amount in the pool box is to gather issued reward tokens from oracles. I think we should explicitly prohibit this path (changing the reward token amount). Since reward tokens are supposed to be exchanged, I don't think "letting them back in" is a good idea.

Reply:
Hmm, now that I think about it I agree. What I had in mind was maybe some oracle pool that has liquidity added to it periodically, say for example something like Night Owl might fund their oracles with Night Owl tokens, but since those are backed by stablecoins, they might not always have enough to keep the pool running indefinitely.

For the time being though, I think this case can be removed/simplified